packages/xstate-wallet/src/chain.ts

```diff
@@ -100,6 +115,35 @@ export class ChainWatcher implements Chain {
     );
   }
 
+  public async ethereumEnable(): Promise<void> {
+    if (window.ethereum) {
+      return await window.ethereum.enable();
+    } else {
+      return Promise.reject('window.ethereum not found');
+    }
+  }
+
+  public get ethereumIsEnabled(): boolean {
+    if (window.ethereum) {
+      return !!window.ethereum.selectedAddress;
+    } else {
+      return false;
+    }
+  }
+
+  public get selectedDestination(): string {
```

Comment:
I think we ought to avoid conflating `destination` (which is a nitro term) and `selectedAddress` (which is an ethereum / MM term), throughout. We should use `selectedAddress` here (downstream code such as the app can decide what to do with it, such as using it as the `destination` in the outcome for a channel.